output/baseoutput/clientsession.go

```diff
@@ -57,11 +57,11 @@ func newClientSession(client *ClientWorker, conn ClientConnection, leftovers cha
 	}
 }
 
-func (session *clientSession) run(maxDuration time.Duration) (chan base.LogChunk, reconnectPolicy) {
+func (session *clientSession) Run(leftovers chan base.LogChunk, maxDuration time.Duration) (chan base.LogChunk, reconnectPolicy) {
 	go session.runAcknowledger()
 
 	// send leftovers from previous sessions
```

Comment:
Could you put the leftovers handling in a submethod? I think that you can get rid of the `break REPLAY_LEFTOVERS` and replace it with a (less jarring) `return` then. 

Also, when you do that, you should put the "normal" forwarding of chunks in its own submethod as well. This would adhere to the [Single level of abstraction principle](http://principles-wiki.net/principles:single_level_of_abstraction).